dask/dataframe/io/tests/test_parquet.py

```diff
@@ -188,6 +188,60 @@ def test_columns_index(tmpdir, write_engine, read_engine, columns, index):
     assert_eq(df[[]], ddf2)
 
 
+def test_index_with_multi_index(tmpdir, engine):
+    fn = os.path.join(str(tmpdir), 'test.parquet')
+    index = pd.MultiIndex.from_arrays([np.arange(10), np.arange(10) + 1],
+                                      names=['x0', 'x1'])
+    df = pd.DataFrame(np.random.randn(10, 2), columns=['a', 'b'], index=index)
+    df2 = df.reset_index(drop=False)
+
+    if engine == 'fastparquet':
+        fastparquet.write(fn, df, write_index=True)
+
+        # fastparquet doesn't support multi-index
+        with pytest.raises(ValueError):
+            ddf = dd.read_parquet(fn, engine=engine)
+    else:
+        import pyarrow as pa
+        pq.write_table(pa.Table.from_pandas(df), fn)
+
+        ddf = dd.read_parquet(fn, engine=engine)
+        assert_eq(ddf, df)
+
+        d = dd.read_parquet(fn, columns=['x0', 'a'], engine=engine)
+        assert_eq(d, df[['a']])
+
+    # DataFrame output
+    d = dd.read_parquet(fn, index=False, engine=engine)
+    assert_eq(d, df2)
+
+    d = dd.read_parquet(fn, index=['a'], engine=engine)
+    assert_eq(d, df2.set_index('a')[['b']])
```

Comment:
Shouldn't `d` contain columns `'x0'` and `'x1'`?

Reply:
Not as currently implemented (on master, or any previous release).

- Data can be an index or a column, with indices always taking preference.
- If `index` is specified, the existing indices are dropped, not added to the set of columns. Indices are only added to the list of columns if `index=False`.

I figured matching the existing behavior made the most sense here, while also making it more robust to multi-indices.

Reply:
OK, so this is the only one left that I cold quibble with - but we ca always tell people to explicitly set `index` and `columns` (or use `index=False` and set index later) if they want those other index columns to come out.

Reply:
This is the only behavior that seems internally consistent. Both columns and index follow the rule "fields use the classification specified by the metadata unless explicitly set as something else".